class-06-ajax-and-json/pair-assignment/starter-code/scripts/article.js

```diff
@@ -50,18 +50,23 @@ Article.fetchAll = function() {
     // When rawData is already in localStorage,
     // we can load it by calling the .loadAll function,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(//TODO: What do we pass in here to the .loadAll function?
-    );
-    articleView.someFunctionToCall//(); //TODO: Change this fake method call to the correct one that will render the index page.
+    Article.loadAll(JSON.parse(localStorage.rawData));
+      //DONE: What do we pass in here to the .loadAll function?
+    articleView.initIndexPage(); //DONE: Change this fake method call to the correct one that will render the index page.
   } else {
-    // TODO: When we don't already have the rawData, we need to:
+    $.getJSON('data/hackerIpsum.json', function ( data ) {
+      Article.loadAll(data);
+      var tempData = JSON.stringify(data);
+      localStorage.setItem('rawData', tempData);
```

Comment:
This is fine for testing, though once you've determined that it's working you don't need that additional variable for `tempData`. It could be refactored/simplified to:
``` javascript
localStorage.setItem('rawData', JSON.stringify(data))
```